amazon/ion/core.py

```diff
@@ -104,16 +104,31 @@ def is_stream_signal(self):
         return self < 0
 
 
-def _scale_to_precision(microseconds, fractional_precision):
-    """Scales the ```fractional_seconds``` attribute by -```fractional_precision```.
+def _scale_to_precision(value, fractional_precision):
+    """Scales the ```value``` attribute by -```fractional_precision```.
+
+    If the length of the value is greater than the ```fractional_precision```, it is determined if the extra digits in
+    ```value``` are not significant, i.e. have the value '0'.
+    If a significant digit is found, a ValueError is raised.
+    If no significant digits are not found, the ```value``` is truncated to match that of the```fractional_precision```
+    and the scaling is performed.
 
     Args:
-        microseconds (int): The numerical value to scale.
+        value (int): The numerical value to scale.
         fractional_precision (int): The negative of this numerical value to scale ```fractional_seconds``` by.
     Returns:
         Decimal: The scaled value.
     """
-    return Decimal(microseconds).scaleb(-fractional_precision)
+    value_string = str(value)
+    if len(value_string) > fractional_precision:
+        for digit in value_string[fractional_precision:]:
+            if digit != '0':
+                raise ValueError('Found timestamp fractional precision of %d digits, '
+                                 'which is less than needed to serialize %d microseconds.'
+                                 % (fractional_precision, value))
+        value_string = value_string[0:fractional_precision]
+        value = int(value_string)
+    return Decimal(value).scaleb(-fractional_precision)
```

Comment:
I went this route because this exact sort of logic using `__str__` was being used in `reader_text.py` initially. The only difference, however, was in the method that used this logic in `reader_text.py` we didn't cast from `Decimal` to string to use this logic, as within the method parameters a string was being passed.